packages/auth/amplify_auth_cognito_dart/lib/src/sdk/sdk_bridge.dart

```diff
@@ -795,77 +797,71 @@ extension MfaSettings on CognitoIdentityProviderClient {
     required String accessToken,
     MfaPreference? sms,
     MfaPreference? totp,
+    MfaPreference? email,
   }) async {
     final UserMfaPreference(
       enabled: currentEnabled,
-      preferred: currentPreference
-    ) = await _getRawUserSettings(
-      accessToken: accessToken,
-    );
+      preferred: currentPreference,
+    ) = await _getRawUserSettings(accessToken: accessToken);
+
+    final newPreferredMethods = [
+      if (sms == MfaPreference.preferred) MfaType.sms,
+      if (totp == MfaPreference.preferred) MfaType.totp,
+      if (email == MfaPreference.preferred) MfaType.email,
+    ];
+
+    if (newPreferredMethods.length > 1) {
+      throw const InvalidParameterException(
+        'Cannot assign multiple MFA methods as preferred',
+      );
+    }
+
+    var preferred = newPreferredMethods.isNotEmpty
+        ? newPreferredMethods.first
+        : currentPreference;
+
+    final isCurrentPreferenceDisabled = switch (currentPreference) {
+      MfaType.sms =>
+        sms == MfaPreference.disabled || sms == MfaPreference.notPreferred,
+      MfaType.totp =>
+        totp == MfaPreference.disabled || totp == MfaPreference.notPreferred,
+      MfaType.email =>
+        email == MfaPreference.disabled || email == MfaPreference.notPreferred,
+      _ => false,
+    };
+    preferred = isCurrentPreferenceDisabled ? null : preferred;
```

Comment:
would you explain the logic behind this line? I can't understand why to set preferred to null when isCurrentPreferenceDisabled given than earlier prefered is set to newPreferredMethods.first if user has a newPreferredMethod.

Reply:
For example, if a user passes in `sms: MfaPreference.disabled` and nothing else. The user's current MfaPreference is `sms: MfaPreference.preferred`. so the `isCurrentPreferenceDisabled` checks to see that if a user disables the current null or makes it not preferred, that the current preferred choice is cleared.